firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/analytics/BlockingAnalyticsEventLogger.java

```diff
@@ -0,0 +1,86 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.crashlytics.internal.analytics;
+
+import android.os.Bundle;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import com.google.firebase.crashlytics.internal.Logger;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Analytics event logger which logs an event to Firebase Analytics using the Crashlytics origin,
+ * and blocks until it receives that event back from the Analytics Event Receiver, or until the
+ * timeout has elapsed.
+ */
+public class BlockingAnalyticsEventLogger implements AnalyticsEventReceiver, AnalyticsEventLogger {
+
+  private static final String APP_EXCEPTION_EVENT_NAME = "_ae";
+
+  private final CrashlyticsOriginAnalyticsEventLogger baseAnalyticsEventLogger;
+  private final int timeout;
+  private final TimeUnit timeUnit;
+
+  private final Object latchLock = new Object();
+
+  private CountDownLatch eventLatch;
+
+  public BlockingAnalyticsEventLogger(
+      @NonNull CrashlyticsOriginAnalyticsEventLogger baseAnalyticsEventLogger,
+      int timeout,
+      TimeUnit timeUnit) {
+    this.baseAnalyticsEventLogger = baseAnalyticsEventLogger;
+    this.timeout = timeout;
+    this.timeUnit = timeUnit;
+  }
+
+  @Override
+  public void logEvent(@NonNull String name, @Nullable Bundle params) {
+    synchronized (latchLock) {
```

Comment:
Is it worth making a comment somewhere that we only ever expect at most ONE of these events per app execution? That fact influences the design, though I think the way you have it implemented is resilient to multiple events (though you could end up with a bunch calls waiting on the syncrhonized(latchLock) block, which is fine).

Maybe not necessary since the design works either way?

Reply:
I think there's definitely the possibility we could end up with multiple queued-up calls to this function, so it may be worth changing this later to allow for sending multiple events at once.